### PR TITLE
Restore VM guest drivers to initramfs file.

### PR DESCRIPTION
**Pull request by cwize1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

In change #8126, the mkinitrd script was removed in favor of using dracut directly. However, one feature that was missed was that the mkinitrd script pulls additional dracut settings from files under the /var/lib/initramfs/kernel directory. This was being used to add the VM guest drivers to initramfs file. Since these drivers are no longer being added, the images don't boot under Hyper-V.

This change re-adds the guest VM drivers to dracut, but using the /etc/dracut.conf.d config directory.

###### Change Log  

- Restore VM guest drivers to initramfs file.

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Manually built image file.

### CODE REVIEW

File: SPECS-EXTENDED/kernel-rt/kernel-rt.spec [hunk 1]
```diff
@@ -421,6 +414,9 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_sysconfdir}/bash_completion.d/bpftool
 
 %changelog
+* Wed Mar 06 2024 Chris Gunn <chrisgun@microsoft.com> - 6.6.7.1-3
+- Remove /var/lig/initramfs/kernel files.
```
Comment (vinceaperri) [inline SPECS-EXTENDED/kernel-rt/kernel-rt.spec:418]: /var/log?

File: SPECS-SIGNED/kernel-signed/kernel-signed.spec [hunk 2]
```diff
@@ -147,6 +146,9 @@ echo "initrd of kernel %{uname_r} removed" >&2
 %exclude /module_info.ld
 
 %changelog
+* Wed Mar 06 2024 Chris Gunn <chrisgun@microsoft.com> - 6.6.7.1-4
+- Remove /var/lig/initramfs/kernel files.
```
Comment (vinceaperri) [inline SPECS-SIGNED/kernel-signed/kernel-signed.spec:150]: /var/log?

File: SPECS/initramfs/initramfs.spec [hunk 3]
```diff
@@ -111,9 +110,11 @@ echo "initramfs" %{version}-%{release} "postun" >&2
 %files
 %defattr(-,root,root,-)
 %{_sysconfdir}/dracut.conf.d/fscks.conf
-%dir %{_localstatedir}/lib/initramfs/kernel
 
 %changelog
+* Wed Mar 06 2024 Chris Gunn <chrisgun@microsoft.com> - 3.0-3
+- Remove /var/lig/initramfs/kernel files.
```
Comment (vinceaperri) [inline SPECS/initramfs/initramfs.spec:116]: /var/log?

File: SPECS/kernel-mshv/kernel-mshv.spec [hunk 4]
```diff
@@ -231,6 +224,9 @@ echo "initrd of kernel %{uname_r} removed" >&2
 %{_includedir}/perf/perf_dlfilter.h
 
 %changelog
+* Wed Mar 06 2024 Chris Gunn <chrisgun@microsoft.com> - 5.15.126.mshv3-7
+- Remove /var/lig/initramfs/kernel files.
```
Comment (vinceaperri) [inline SPECS/kernel-mshv/kernel-mshv.spec:228]: /var/log?

File: SPECS/kernel/kernel.spec [hunk 5]
```diff
@@ -416,6 +409,9 @@ echo "initrd of kernel %{uname_r} removed" >&2
 %{_sysconfdir}/bash_completion.d/bpftool
 
 %changelog
+* Wed Mar 06 2024 Chris Gunn <chrisgun@microsoft.com> - 6.6.14.1-4
+- Remove /var/lig/initramfs/kernel files.
```
Comment (vinceaperri) [inline SPECS/kernel/kernel.spec:413]: /var/log?

File: SPECS-EXTENDED/kernel-rt/kernel-rt.spec [hunk 6]
```diff
@@ -421,6 +414,9 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_sysconfdir}/bash_completion.d/bpftool
 
 %changelog
+* Wed Mar 06 2024 Chris Gunn <chrisgun@microsoft.com> - 6.6.7.1-3
+- Remove /var/log/initramfs/kernel files.
```
Comment (vinceaperri) [inline SPECS-EXTENDED/kernel-rt/kernel-rt.spec:418]: Actually I think it should be /var/lib. Sorry!

**Review by vinceaperri — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]
I built a local VHDX with these changes and confirmed the image boots with hyperv

File: SPECS/dracut/dracut.spec [hunk 7]
```diff
@@ -135,6 +132,7 @@ install -m 0644 dracut.conf.d/fips.conf.example %{buildroot}%{_sysconfdir}/dracu
 > %{buildroot}%{_sysconfdir}/system-fips
 
 install -m 0644 %{SOURCE3} %{buildroot}%{_sysconfdir}/dracut.conf.d/50-megaraid.conf
+install -m 0644 %{SOURCE6} %{buildroot}%{_sysconfdir}/dracut.conf.d/00-defaults.conf
```
Comment (christopherco) [inline SPECS/dracut/dracut.spec:135]: Just curious, is 00-defaults.conf the right importance/ordering value to use here?

Comment (cwize1) [reply SPECS/dracut/dracut.spec:135]: No idea. Considering that these are default options for the image that can be overridden by the user, `00` made sense to me.

**Review by christopherco — APPROVED** [review]

**Pull request merged**